js/compiler/enhanced.go

```diff
@@ -0,0 +1,55 @@
+package compiler
+
+import (
+	"path/filepath"
+
+	"github.com/dop251/goja/file"
+	"github.com/dop251/goja/parser"
+	"github.com/evanw/esbuild/pkg/api"
+)
+
+func esbuildTransform(src, filename string) (code string, srcMap []byte, err error) {
+	opts := api.TransformOptions{
+		Sourcefile:     filename,
+		Loader:         api.LoaderJS,
+		Target:         api.ES2017,
```

Comment:
This makes esbuild unneedingly transform a whole bunch of code that goja supports.

In practice from the list of features on https://esbuild.github.io/content-types/#javascript 

The only ones not supported are async generators, async iteration and  `??=`. And in my test the async generator and async iteration doesn't work with the example from [mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function*).

This also adds a bunch of esnext stuff which I haven't tested, but will argue are probably even less desired by even our less average users given that they aren't released yet.

If you will likely to enable any of those it will be nice to also enable their tc39 tests respectfully. I would recommend that this is done *after* the initial release of this instead of now.

So my strong recommendation is this to be set to ESNext - which in practice disables esbuild doing any transformation for features and leave just the TypeScript + commonJS transofmration. Some tc39 tests get slightly different results witht that.

You can use the `Supported` field to set which transformation you want to add. Again I will recommend doing this after this is merged, maybe even after we have had some feedback.